addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java

```diff
@@ -254,9 +260,40 @@ public boolean nativePing(IpPingMethodEnum method, String hostname, int timeoutI
 
         }
 
-        // The return code is 0 for a successful ping. 1 if device didn't respond and 2 if there is another error like
-        // network interface not ready.
-        return proc.waitFor() == 0;
+        // The return code is 0 for a successful ping, 1 if device didn't
+        // respond, and 2 if there is another error like network interface
+        // not ready.
+        // Exception: return code is also 0 in Windows for all requests on the local subnet.
+        // see https://superuser.com/questions/403905/ping-from-windows-7-get-no-reply-but-sets-errorlevel-to-0
+        if (method != IpPingMethodEnum.WINDOWS_PING) {
+            return proc.waitFor() == 0;
+        }
+
+        int result = proc.waitFor();
+        if (result != 0) {
+            return false;
+        }
+
+        try (BufferedReader r = new BufferedReader(new InputStreamReader(proc.getInputStream()))) {
+            String line = r.readLine();
+            if (line == null) {
+                logger.trace("Received no output from ping process.");
```

Comment:
This should never happen, should it? I'd say the logger level might be raised to a warning, or alternatively throw an IOException.

Reply:
I removed the logging and put in throws.